### PR TITLE
Fix for Incorrect warning log about deprecated options hostname, hostname-debug

### LINKED ISSUE #30380: Incorrect warning log about deprecated options hostname, hostname-debug
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

After updating to 25.0.0 and changing the configuration to the new hostname v2 options, a warning is logged on startup:
```
2024-06-12 12:21:17,077 WARN  [org.keycloak.quarkus.runtime.cli.Picocli] (main) The following used options or option values are DEPRECATED and will be removed or their behaviour changed in a future release:
        - hostname-debug
        - hostname
Consult the Release Notes for details.
```

The feature hostname v2 is enabled (by default) and hostname v1 is disabled:

image

The new configuration also works as expected:

image

It just seems like the new options are incorrectly marked as deprecated.

### Version

25.0.0

### Regression

- [ ] The issue is a regression

### Expected behavior

No warnings about deprecated options are logged when using hostname v2 options.

### Actual behavior

A warning about deprecated options is logged on startup.

### How to Reproduce?

1. Use [hostname v2 configuration options](https://www.keycloak.org/server/hostname), e.g. with
    ```env
    KC_HOSTNAME=https://my.keycloak.hostname
    KC_HOSTNAME_DEBUG=true
    ```
2. Start Keycloak and observe the first log line

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kaustubh-rh**
Added a check in handleDeprecated function in Picocli.java
Closes #30380

### CODE REVIEW

**Comment by kaustubh-rh** [comment]
Hi @vmuzikar , raised a PR for the same :)

**Comment by mposolda** [comment]
@vmuzikar Assigning PR to you as the issue itself is also assigned to you.

File: quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java [hunk 1]
```diff
@@ -466,29 +465,35 @@ private static void handleDeprecated(Set<String> deprecatedInUse, PropertyMapper
...
 
-        StringBuilder sb = new StringBuilder("\t- ");
-        sb.append(optionName);
+        StringBuilder sb = new StringBuilder();
+        if (mapper.getDeprecatedMetadata().isPresent()) {
 
-        if (!deprecatedValuesInUse.isEmpty()) {
-            sb.append("=").append(String.join(",", deprecatedValuesInUse));
-        }
+            if (!mapper.getDeprecatedMetadata().get().getNewOptionsKeys().isEmpty() || mapper.getDeprecatedMetadata().get().getNote() != null || !mapper.getDeprecatedMetadata().get().getDeprecatedValues().isEmpty()) {
```
Comment (vmuzikar) [inline quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java:471]: I'm afraid we can't do this. With this condition that don't have any new keys, note, or value will not be printed. A warning should be emitted when e.g. Keycloak is started as:
```
kc.sh start-dev --features=hostname:v1 --hostname=abc
```

**Review by vmuzikar — CHANGES_REQUESTED** [review]
@kaustubh-rh Thank you for the PR!

**Pull request closed without merging**